VSShortcutsManager/AddNewShortcut/AddKeyboardShortcut.xaml.cs

```diff
@@ -23,26 +24,109 @@ public partial class AddKeyboardShortcut : Window
     {
 
         private IServiceProvider _serviceProvider;
+        private VSShortcutQueryEngine _queryEngine;
         public AddKeyboardShortcut(IServiceProvider serviceProvider)
         {
             InitializeComponent();
             _serviceProvider = serviceProvider;
+            _queryEngine = new VSShortcutQueryEngine(_serviceProvider);
             cmbCommandList.ItemsSource = new CommandListViewModel(_serviceProvider).DataSource;
             cmbScopeList.ItemsSource = new ScopeListViewModel(_serviceProvider).DataSource;
         }
         private void btnClose_Click(object sender, RoutedEventArgs e) => this.Close();
 
         private void btnAddShortcut_Click(object sender, RoutedEventArgs e)
         {
-            string shortcutcommand = cmbCommandList.SelectedValue.ToString();
-            string shortcutScope = cmbScopeList.SelectedValue.ToString();
-            string shortcutKeys = txtShortcut.Text;
-            string shortcutBinding  = shortcutScope + "::" + shortcutKeys;
-           
-            VSShortcutQueryEngine queryEngine = new VSShortcutQueryEngine(_serviceProvider);
-           
+            try
+            {
+                string shortcutcommand = cmbCommandList.SelectedValue.ToString();
+                string shortcutScope = cmbScopeList.SelectedValue.ToString();
+                string shortcutKeys = txtShortcut.Text;
+                string shortcutBinding = shortcutScope + "::" + shortcutKeys;
+                //Check if potential conflicts available
+                if (listConflicts.Items.Count > 0)
+                {
+                    if (seekConfirm(shortcutScope, shortcutKeys) == MessageBoxResult.Yes)
+                    {
+                        _queryEngine.BindShortcut(shortcutcommand, shortcutBinding);
```

Comment:
This code is duplicated a few lines below. To avoid duplication, I would have written the first two IFs with confirm aborting function if user says no.